### ISSUE TITLE
Accessibility: Error messages on Lost Password form not announced to screen readers

### ISSUE THREAD

**Issue by amberhinds** [woothemes/woocommerce#46559]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

On the password reset form if the form is submitted without filling in a username and password, there is a visible error "Enter a username or email address." that is not announced to screen reader users.
<img width="1659" alt="Screenshot error message" src="https://github.com/woocommerce/woocommerce/assets/13153456/d123fd18-dfa7-4f01-a8eb-e96df25fae46">

This is related to https://github.com/woocommerce/woocommerce/issues/43659 and https://github.com/woocommerce/woocommerce/issues/43635

Essentially, though this element has a role="alert" on it, it's not announced because the alert role is only intended for dynamic elements, and this form does not submit or validate dynamically, only on page load. 

The alert role is not read out on page load only on dynamic elements. [Read more on Mozilla Developer Docs ](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/alert_role)

### Expected behavior

When a form error is present, there should be an auditory message for screen reader users to know that there was an error.

You can either do this by:
A) When the page reloads, set keyboard focus on the error message so it is read out. OR
B) Add an [aria-live](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-live) region on the status message and trigger it **after** the page loads.

You may also consider changing this form to submit with ajax so screen reader users are not sent to the top of the page as happens with a refresh and the existing alert is announced.

### Actual behavior

The error is not announced by the screen reader.

### Steps to reproduce

1. Turn on a screen reader such as VoiceOver or NVDA.
2. Go to the Lost Password page (/my-account/lost-password).
3. Do not fill in the username or email address field, but submit the form.
4. Listen as the page reloads and observe there's no announcement about the error.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.
